app/grandchallenge/evaluation/migrations/0073_migrate_interfaces_for_phases.py

```diff
@@ -0,0 +1,52 @@
+from django.db import migrations
+
+from grandchallenge.algorithms.models import (
+    get_existing_interface_for_inputs_and_outputs,
+)
+from grandchallenge.evaluation.utils import SubmissionKindChoices
+
+
+def add_algorithm_interfaces_for_phases(apps, _schema_editor):
+    Phase = apps.get_model("evaluation", "Phase")  # noqa: N806
+    AlgorithmInterface = apps.get_model(  # noqa: N806
+        "algorithms", "AlgorithmInterface"
+    )
+
+    for phase in (
+        Phase.objects.filter(submission_kind=SubmissionKindChoices.ALGORITHM)
+        .prefetch_related("algorithm_inputs", "algorithm_outputs")
+        .all()
+    ):
+        inputs = phase.algorithm_inputs.all()
+        outputs = phase.algorithm_outputs.all()
+
+        if not inputs or not outputs:
+            # Skip phases without inputs or outputs
+            continue

```

Comment:
I am unsure if this is necessary.

Reply:
But that would mean that there is a borked configuration in the database, so should be fixed before migrating rather than just silently skipping over it, no?

Reply:
Not sure. It could be phases that are in the middle of set-up - the BigPicture PoC for example. I think it's fine to silently skip over these since we are the ones to configure the phases (not the challenge admins) and if they are not fully set-up yet, we can do it later from the UI rather than postponing the migration until the onboarding is complete, right? 

We could do a quick check in the shell to see how many phases this concerns. In theory, it should not happen and both inputs and outputs should be set-up at the same time, but it is theoretically possible (see DemoBP).

Reply:
In the next migration you will presumably remove the old fields though, so that information would anyway be lost.

Reply:
Discussed this synchronously, I will raise an error here rather than continue. I have also added a todo item to the pitch to check if there are any algorithms or phases with incomplete set-up.